app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -382,6 +383,26 @@ def save_labels_inventory(entity, hashes, target = nil)
     store_ids_for_new_records(entity.labels, hashes, [:section, :name])
   end
 
+  def save_labels_and_tags_inventory(entity, hashes, target = nil)
+    save_labels_inventory(entity, hashes, target)
+    save_tags_generated_from_labels(entity, hashes, target)
+  end
+
+  # Must be called after object already got its labels.
+  # This is achieved by :labels_and_tags going through save_labels_and_tags_inventory().
+  # TODO: Drop the order requirement by using the :labels hash instead of entity.labels?
+  #   Would still need still save_labels_and_tags_inventory() to get that hash.
+  def save_tags_generated_from_labels(entity, _hashes, _target = nil)
+    current_tags = ContainerLabelTagMapping.tags_for_entity(entity)
+    possible_old_tags = ContainerLabelTagMapping.all_mapped_tags
```

Comment:
- I would name this variable mappable_tags. 
- Moving forward we will need to make sure mappable tags cannot be assigned by the user so we do not delete user assigned tags on refreshes